hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/OnJoinOp.java

```diff
@@ -90,6 +94,18 @@ public void run() throws Exception {
                     getLogger().warning("Error while running post-join operation: " + op, e);
                 }
             }
+
+            final ClusterService clusterService = getService();
```

Comment:
I have seen that this line causing NPE in some tests on maintenance branch. @petrpleshachkov

Reply:
Indeed, for instance here: http://jenkins.hazelcast.com/job/Hazelcast-EE-pr-builder/1707/testReport/junit/com.hazelcast.spi.hotrestart.cluster/HotRestartDistributedObjectsTest/test_getDistributedObjects_afterHotRestartProcessCompletes/

Full stacktrace:
```
java.lang.NullPointerException
	at java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936) ~[?:1.8.0_201]
	at com.hazelcast.spi.impl.servicemanager.impl.ServiceManagerImpl.getServiceInfo(ServiceManagerImpl.java:343) ~[classes/:?]
	at com.hazelcast.spi.impl.servicemanager.impl.ServiceManagerImpl.getService(ServiceManagerImpl.java:364) ~[classes/:?]
	at com.hazelcast.spi.impl.NodeEngineImpl.getService(NodeEngineImpl.java:364) ~[classes/:?]
	at com.hazelcast.spi.Operation.getService(Operation.java:395) ~[classes/:?]
	at com.hazelcast.internal.cluster.impl.operations.OnJoinOp.run(OnJoinOp.java:98) ~[classes/:?]
	at com.hazelcast.spi.Operation.call(Operation.java:170) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:210) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:199) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:407) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.run(OperationServiceImpl.java:287) [classes/:?]
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.finalizeJoin(ClusterServiceImpl.java:397) [classes/:?]
	at com.hazelcast.internal.cluster.impl.operations.FinalizeJoinOp.run(FinalizeJoinOp.java:88) [classes/:?]
	at com.hazelcast.spi.Operation.call(Operation.java:170) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:210) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:199) [classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:416) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:153) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:123) [classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110) [classes/:?]
```